src/Control/Controller.php

```diff
@@ -528,9 +532,16 @@ public function render($params = null)
      * Call this to disable site-wide basic authentication for a specific controller. This must be
      * called before Controller::init(). That is, you must call it in your controller's init method
      * before it calls parent::init().
+     *
+     * @deprecated 4.0..5.0
      */
     public function disableBasicAuth()
     {
+        Deprecation::notice(
+            '5.0',
+            'Add this controller\'s url to SilverStripe\Security\BasicAuthMiddleware.ExcludeURLPatterns'
+            . ' property instead'
```

Comment:
Can we avoid the use of a hardcoded class? I'd prefer `BasicAuthMiddleware::class . 'ExcludeURLPatterns'`